CesiumAsync/include/CesiumAsync/AsyncSystem.h

```diff
@@ -15,6 +15,91 @@ class ITaskProcessor;
 
 template <typename T> class Future;
 
+class AsyncSystem;
+
+/**
+ * @brief A promise that can be resolved or rejected by an asynchronous task.
+ *
+ * @tparam T The type of the object that the promise will be resolved with.
+ */
+template <typename T> class Promise {
```

Comment:
Nitpick from me, should we separate this out into another file, similar to Future?

Reply:
Good idea, done.